Appendix B suggests signals for a DMI with one DTM connected to one DM

kmeinhar opened the pull request:
Closes #523

timsifive commented:
In that case, can we also add a line with a link to the Rocket Chip implementation? It'll be good for people to know

ernie-sifive commented on debugger_implementation.tex:
```diff
@@ -33,6 +33,39 @@ \section{Debug Module Interface Access} \label{dmiaccess}
+
+As stated in section \ref{dmi} the details of the DMI is left to the system designer.
+It is quite often the case that only one DTM and DM is implemented.
+In this case it might be useful to comply with the signals suggested in table \ref{tab:dmi_signals}.
```
```suggestion
In this case it might be useful to comply with the signals suggested in table \ref{tab:dmi_signals},
which is the implementation used in the open-source rocket-chip RISC-V core:
https://github.com/sifive/rocket-chip/blob/master/src/main/scala/devices/debug/Debug.scala#L170
```

kmeinhar commented:
I added 4 commits that implement the suggested changes.

I like to do a lot of commits, but I know this is not welcomed by everyone.
So if you want I can squash those commits into a single one.

This MR also does not include the recompiled draft PDF.
I would do that if all issues are resolved, when desired.

timsifive commented:
> I like to do a lot of commits, but I know this is not welcomed by everyone.
> So if you want I can squash those commits into a single one.

More commits is great. Thanks.

> This MR also does not include the recompiled draft PDF.

Because git doesn't do a great job handling PDFs, I recompile and submit the PDF myself once a month, and ignore it the rest of the time.

timsifive commented on debugger_implementation.tex:
```diff
@@ -33,6 +33,48 @@ \section{Debug Module Interface Access} \label{dmiaccess}
 
+\section{Debug Module Interface Signals} \label{dmi_signals}
+
+As stated in section \ref{dmi} the details of the DMI is left to the system designer.
```
details of the DMI *are* left

timsifive commented on debugger_implementation.tex:
```diff
@@ -33,6 +33,48 @@ \section{Debug Module Interface Access} \label{dmiaccess}
+which is the implementation used in the open-source \href{https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/devices/debug/Debug.scala#L170}{rocket-chip} RISC-V core.
+
+In order for the DTM to perform a read it has to wait for REQ\_READY to be high.
+Is this the case REQ\_OP can be set to 1 to indicate a read action and REQ\_ADDRESS to the required address.
```
*When* this is the case

timsifive commented on debugger_implementation.tex:
```diff
@@ -33,6 +33,48 @@ \section{Debug Module Interface Access} \label{dmiaccess}
+Is this the case REQ\_OP can be set to 1 to indicate a read action and REQ\_ADDRESS to the required address.
+Finally REQ\_VALID is set high, indicating to the DM that a valid request is pending.
+
+The DM can respond to a request from the DTM when RSP\_READY is high.
```
The DM *must* respond
Unless you want to give it an option not to respond, in which case *should* is probably the right term.

timsifive commented on debugger_implementation.tex:
```diff
@@ -33,6 +33,48 @@ \section{Debug Module Interface Access} \label{dmiaccess}
+Finally REQ\_VALID is set high, indicating to the DM that a valid request is pending.
+
+The DM can respond to a request from the DTM when RSP\_READY is high.
+The status of the respond is indicated with the RSP\_OP signal (See \FdtmDmiOp) and potential respond data with RSP\_DATA.
```
status of the *response* is indicated *by* the RSP\_OP signal (*see \FdtmDmiOp). The data of the response is driven to RSP\_DATA.

kmeinhar commented:
Moved to the hardware implementation chapter.

Fixed Spelling mistakes.

Rephrased the paragraph on DTM request to include read and write requests.

kmeinhar commented:
Thank you very much for your support and feedback to get the MR into shape!
